src/boundingbox.h

```diff
@@ -44,9 +47,31 @@ BoundingBox<T>::BoundingBox(std::initializer_list<typename BoundingBox::Interval
   if (intervals.size() != dimensions_)
     throw RStarException("Dimensions mismatch");
   for (const Interval &interval : intervals)
-    intervals_[i] = interval;
+    {
+        intervals_[i] = interval;
+        ++i;
+    }
+}
+
+template <typename T>
+typename BoundingBox<T>::CoordType BoundingBox<T>::HyperArea() const
+{
+    //if (BoundingBox<T>::Intervals.size() == 0)
+    //    throw RStarException("No dimensions found");
+    size_t i = 0;
+    BoundingBox<T>::CoordType res = 1;
+    for (const typename BoundingBox<T>::Interval &interval : BoundingBox<T>::intervals_)
+    {
+        res *= ( std::get<1>(GetInterval(i)) ) - ( std::get<0>(GetInterval(i)) );
+        //std::cout << std::get<0>(GetInterval(i)) << std::endl;
+        ++i;
+    }
+    //std::cout << res << std::endl;
+    return res;
```

Comment:
guardar `res` en una caché, digamos

    return this->area_ = res;

Dar valor negativo inicial para indicar que no se ha calculado el área aún. Descartar área cuando hay cambios del boundingbox (cuando hay llamadas a los setters)